cfme/markers/upgrade.py

```diff
@@ -0,0 +1,60 @@
+"""Marker definitions for upgrade testing"""
+import pytest
+
+
+# Add Marker
+def pytest_configure(config):
+    config.addinivalue_line("markers", "post_upgrade: Mark test for post upgrade testing")
+
+
+# Filtering options
+def pytest_addoption(parser):
+    group = parser.getgroup("cfme")
+    group.addoption(
+        "--upgrade-appliance",
+        dest="upgrade_appliance",
+        action="store_true",
+        default=False,
+        help="Enable upgrade an appliance before the first test is run.",
+    )
+    group.addoption(
+        "--upgrade-to",
+        dest="upgrade_to",
+        action="store",
+        default="5.11.z",
+        help="Supported versions 5.9.z, 5.10.z, 5.11.z (.z means latest and default is 5.11.z)",
+    )
+
+
+# Test collection modifier
+def pytest_collection_modifyitems(session, config, items):
+    """Provide filtering of test case collection based on the CLI options"""
+    post_upgrade = config.getoption("upgrade_appliance")
+
+    if not post_upgrade:
+        return
+
+    keep, discard = [], []
+
+    for item in items:
+        # Filter test as per marker
+        if post_upgrade and item.get_marker("post_upgrade"):
```

Comment:
No need to check post_upgrade again, at this point the function should have returned if it was not set.

Reply:
right :+1: